### PR TITLE
feat: localized field name based on format specifier

### PR DESCRIPTION
**Pull request by ggkrustev**
### Description
The purpose of the feature is to get a localized filed name that could replace a specific date format segment:
```
dateFormatPlaceholder("d") -> 'day';
dateFormatPlaceholder("MMMM") -> 'month';
```

### CODE REVIEW

**Comment by ggkrustev** [comment]
Accept suggestions for better naming :) The current name is `dateFormatPlaceholder`.

File: src/dates.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
 export { default as formatDate } from './dates/format-date';
 export { default as parseDate } from './dates/parse-date';
-export { default as dateFormatString } from './dates/date-format-string';
\ No newline at end of file
+export { default as dateFormatString } from './dates/date-format-string';
```
Comment (rusev) [inline src/dates.js]: Don't forget to re-export these functions from `kendo-angular-intl` because of CDN bundle. :)

Comment (ggkrustev) [inline src/dates.js]: I won't :) We need to accept the changes here before move to the `kendo-angular-intl` repo anyway

File: src/dates/date-format-placeholder.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+const ZONE = 'zone';
+const WEEKDAY = 'weekday';
+const QUARTER = 'quarter';
+
+const dateFieldMap = {
+    'y': 'year',
+    'M': MONTH,
+    'L': MONTH,
+    'd': DAY,
+    'E': DAY,
```
Comment (Xizario) [inline src/dates/date-format-placeholder.js]: 'E' should be mapped to `WEEKDAY`

**Comment by danielkaradachki** [comment]
How about dateFieldName?

Also, I am not sure how exactly will the function be used, but passing options rather than format specifier seems better from user perspective.

**Comment by ggkrustev** [comment]
@danielkaradachki, I am OK with `dateFieldName`. If no one disagrees, I'll change the function name.

As to the idea for 'passing options instead of format' could you provide an example? Do you imagine smth like that:
```
dateFieldName({ field: 'year', locale: 'en' });
```

File: src/dates/date-format-placeholder.js (hunk 2 above)
Comment (ggkrustev) [inline src/dates/date-format-placeholder.js]: I tend to agree here, although `weekday` will return smth like `day of the week`, which is a bit longer to display in the input. Nevertheless, it's semantically correct I think.
Will change it to 'weekday'.

**Comment by danielkaradachki** [comment]
Yep, something similar, but for consistency I think that the locale should be passed as second parameter.

**Comment by ggkrustev** [comment]
Yep, I see. Although, I see your points about the method consistency, I'll leave the method signature unchanged. Will try to use it in place, before introduce any changes in the arguments.

**Comment by Xizario** [comment]
The most convenient solution for the me would be method like:
```
dateFieldName({ specifier: 'M', format:'d', locale: 'en' }); //returns "mo."
dateFieldName({ specifier: 'M', format:'D', locale: 'en'  }); //returns "month"
dateFieldName({ specifier: 'M', format:'yyyy-MMM-dd', locale: 'en'  }); //returns "month"
```

**Comment by ggkrustev** [comment]
Few changes were introduced in the `dateFieldName` function due to the new [`splitFormatMethod`](https://github.com/telerik/kendo-intl/pull/18) method. It works with DateFieldNameOptions instead of format specifier string.

File: src/cldr.d.ts [hunk 3]
```diff
@@ -33,6 +33,51 @@ export interface DateFormatNameOptions {
 export function dateFormatNames(locale: string, options: DateFormatNameOptions): any;
 
 /**
+ * Settings for the dateFieldName function.
+ */
+export interface DateFieldNameOptions {
+    /**
+     * Specifies the type of names.
+     */
+    type: 'dayPeriods'  |
```
Comment (danielkaradachki) [inline src/cldr.d.ts]: This should be removed, there is already corresponding dayperiod type.

**Comment by ggkrustev** [comment]
@danielkaradachki nope, the method is documented:
https://github.com/telerik/kendo-intl/blob/local-format-datefields/src/cldr.d.ts#L77

**Pull request merged**